pkg/kubectl/cmd/delete.go

```diff
@@ -95,8 +96,37 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return err
 	}
 
+	// By default use a reaper to delete all related resources.
+	if cmdutil.GetFlagBool(cmd, "recursive") {
+		return ReapResult(r, f, out)
+	}
+	return DeleteResult(r, out)
+}
+
+func ReapResult(r *resource.Result, f *cmdutil.Factory, out io.Writer) error {
+	found := 0
+	err := r.Visit(func(info *resource.Info) error {
```

Comment:
Delete must ignore not found, since it can take duplicate arguments.  Reapers need to tolerate not found as well.

> On Apr 23, 2015, at 8:00 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/delete.go:
> 
> > @@ -95,8 +96,37 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
> >  		return err
> >  	}
> >  
> > +	// By default use a reaper to delete all related resources.
> > +	if cmdutil.GetFlagBool(cmd, "recursive") {
> > +		return ReapResult(r, f, out)
> > +	}
> > +	return DeleteResult(r, out)
> > +}
> > +
> > +func ReapResult(r *resource.Result, f *cmdutil.Factory, out io.Writer) error {
> > +	found := 0
> > +	err := r.Visit(func(info *resource.Info) error {
> Looks like DeleteResult is ignoring NotFound errors. Should this do the same? I noticed that someReapers are aggressively getting objects before deletion.
> 
> —
> Reply to this email directly or view it on GitHub.
>